src/AspNet.Security.OAuth.Moodle/MoodleAuthenticationOptions.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using static AspNet.Security.OAuth.Moodle.MoodleAuthenticationConstants;
+
+namespace AspNet.Security.OAuth.Moodle
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="MoodleAuthenticationHandler"/>.
+    /// </summary>
+    public class MoodleAuthenticationOptions : OAuthOptions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MoodleAuthenticationOptions"/> class.
+        /// </summary>
+        public MoodleAuthenticationOptions()
+        {
+            AuthorizationEndpoint = MoodleAuthenticationDefaults.AuthorizationEndpoint;
+            CallbackPath = MoodleAuthenticationDefaults.CallbackPath;
+            TokenEndpoint = MoodleAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = MoodleAuthenticationDefaults.UserInformationEndpoint;
+
+            Scope.Add("user_info");
+
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "username");
+
+            ClaimActions.MapCustomJson(ClaimTypes.Name,
+                e =>
+                    e.GetString("lang")?.StartsWith("zh", System.StringComparison.InvariantCultureIgnoreCase) ?? false ?
+                        $"{e.GetString("lastname")}{e.GetString("firstname")}" :
```

Comment:
Should there be a space between the two names? I'm not familiar with exactly how names in China should be written natively.

Would be one of the following based on your answer:

```csharp
e.GetString("lastname") + e.GetString("firstname")
```

or:

```csharp
e.GetString("lastname") + " " + e.GetString("firstname")
```